nomad/structs/structs.go

```diff
@@ -3518,6 +3638,18 @@ type Allocation struct {
 	// PreviousAllocation is the allocation that this allocation is replacing
 	PreviousAllocation string
 
+	// DeploymentID identifies an allocation as being created from a
+	// particular deployment
+	DeploymentID string
```

Comment:
Do we actually use this? Since I'm guessing we can GC the Deployments whenever, it may be a dangling pointer most of the time.